src/averageAges.js

```diff
@@ -37,7 +46,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => person.sex === 'f');
+
+  const womenLifespans = women.reduce((totalWomenLifespan, woman) =>
+    (withChildren ? people.some(person => person.mother === woman.name) : true)
+      ? (totalWomenLifespan.push(woman.died - woman.born), totalWomenLifespan)
+      : totalWomenLifespan,
+  []);
```

Comment:
please filter women with `withChildren` argument and use simple `reduce` method without `push`